util/topgen/intermodule.py

```diff
@@ -0,0 +1,111 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+from typing import Dict
+
+from .lib import *
+
+
+def elab_intermodule(topcfg):
+    """Check the connection of inter-module and categorize them
+
+    In the top template, it uses updated inter_module fields to create
+    connections between the modules (incl. memories). This function is to
+    create and check the validity of the connections `inter_module` using IPs'
+    `inter_signal_list`.
+    """
+
+    list_of_intersignals = []
+
+    if "inter_signal" not in topcfg:
+        topcfg["inter_signal"] = {}
+
+    # Gather the inter_signal_list
+    instances = topcfg["module"] + topcfg["memory"]
+
+    intermodule_instances = [x for x in instances if "inter_signal_list" in x]
+
+    for x in intermodule_instances:
+        for sig in x["inter_signal_list"]:
+            # Add instance name to the entry and add to list_of_intersignals
+            sig["inst_name"] = x["name"]
+            list_of_intersignals.append(sig)
+
+    # TODO: Cross check
+
+    # Add field to the topcfg
+    topcfg["inter_signal"]["signals"] = list_of_intersignals
+
+    # intermodule
+    definitions = []
+
+    uid = 0  # Unique connection ID across the top
+
+    for req, rsps in topcfg["inter_module"].items():
+        log.info("{req} --> {rsps}".format(req=req, rsps=rsps))
+        req_module, req_signal = req.split('.')
+
+        # get the module signal
+        req_struct = find_intermodule_signal(list_of_intersignals, req_module,
+                                             req_signal)
+
+        rsp_len = len(rsps)
+        for i, rsp in enumerate(rsps):
+            assert i == 0, "Handling multiple connections (array) isn't yet supported"
+            rsp_module, rsp_signal = rsp.split('.')
+
+            rsp_struct = find_intermodule_signal(list_of_intersignals,
+                                                 rsp_module, rsp_signal)
+
+            # determine the signal name
+            sig_name = "im{uid}_{req_s}".format(uid=uid,
```

Comment:
this is more of a subjective thing, but i tend to like to uniquify with source / destination name pairs instead of a generic unique identifier.. it can get long, but i favor it because i get information from the name immediately.

Don't feel compelled to change though, just a thought.

Reply:
Yeah I agree with it. I will change it.